subsys/bluetooth/host/l2cap.c

```diff
@@ -2361,6 +2361,21 @@ static int l2cap_le_connect(struct bt_conn *conn, struct bt_l2cap_le_chan *ch,
 
 	ch->chan.psm = psm;
 
+	if (!IS_ENABLED(CONFIG_BT_CONN_DISABLE_SECURITY) &&
+	    conn->sec_level < ch->chan.required_sec_level) {
+		int err;
+
+		err = bt_conn_set_security(conn, ch->chan.required_sec_level);
+		if (err) {
+			return err;
+		}
+
+		atomic_set_bit(ch->chan.status,
+			       BT_L2CAP_STATUS_ENCRYPT_PENDING);
+
+		return 0;
+	}
+
```

Comment:
Im fine with this change, though I didn't really consider CONFIG_BT_CONN_DISABLE_SECURITY to be taken into account for client APIs just for server.

Reply:
Ok. I'll remove CONFIG_BT_CONN_DISABLE_SECURITY here then.